druid/src/widget/split.rs

```diff
@@ -0,0 +1,220 @@
+
+use crate::kurbo::{Line, Point, Rect, Size};
+use crate::widget::flex::Axis;
+use crate::{
+    theme, BaseState, BoxConstraints, Data, Env, Event, EventCtx, LayoutCtx, PaintCtx,
+    RenderContext, UpdateCtx, Widget, WidgetPod,
+};
+
+///An arbitrarily chosen constant. We may want to change this later, or make it a field.
+const SPLITTER_SIZE: f64 = 10.0;
+
+///A container containing two other widgets, splitting the area either horizontally or vertically.
+pub struct Split<T: Data> {
+    split_point: f64,
+    split_direction: Axis,
+    child1: WidgetPod<T, Box<dyn Widget<T>>>,
+    child2: WidgetPod<T, Box<dyn Widget<T>>>,
+}
+
+/// A builder for a widget which divides space horizontally
+/// The splitter bar will run top-to-bottom
+///
+/// The actual widget it implemented by [`Split`], but this builder
+/// is provided for convenience.
+///
+/// [`Split`]: struct.Split.html
+pub struct HorizontalSplit;
+impl HorizontalSplit {
+    /// Create a new split panel which splits the horizontal area,
+    /// leaving the full height available to both children.
+    ///
+    /// Note that `split_point` must be between 0 and 1  
+    pub fn new<T: Data>(
+        split_point: f64,
+        child1: impl Widget<T> + 'static,
+        child2: impl Widget<T> + 'static,
+    ) -> Split<T> {
+        Split::new(Axis::Horizontal, split_point, child1, child2)
+    }
+}
+/// A builder for a widget which divides space vertically
+/// The splitter bar will run left-to-right
+///
+/// The actual widget it implemented by [`Split`], but this builder
+/// is provided for convenience.
+///
+/// [`Split`]: struct.Split.html
+pub struct VerticalSplit;
+impl VerticalSplit {
+    /// Create a new split panel which splits the vertical area,
+    /// leaving the full width available to both children.
+    ///
+    /// Note that `split_point` must be between 0 and 1  
+    pub fn new<T: Data>(
+        split_point: f64,
+        child1: impl Widget<T> + 'static,
+        child2: impl Widget<T> + 'static,
+    ) -> Split<T> {
+        Split::new(Axis::Vertical, split_point, child1, child2)
+    }
+}
+impl<T: Data> Split<T> {
+    ///Create a new split panel.
+    ///Note that split_point must be between 0 and 1.
+    fn new(
+        split_direction: Axis,
+        split_point: f64,
+        child1: impl Widget<T> + 'static,
+        child2: impl Widget<T> + 'static,
+    ) -> Split<T> {
+        assert!(
+            split_point > 0.0 && split_point < 1.0,
+            "Invalid split point {} provided!",
+            split_point
+        );
+        Split {
+            split_point,
+            split_direction,
+            child1: WidgetPod::new(child1).boxed(),
+            child2: WidgetPod::new(child2).boxed(),
+        }
+    }
+}
+impl<T: Data> Widget<T> for Split<T> {
+    fn paint(&mut self, paint_ctx: &mut PaintCtx, base_state: &BaseState, data: &T, env: &Env) {
+        let size = base_state.size();
+        let line1;
+        let line2;
```

Comment:
this doesn't feel super 'rusty'. Unless there's a clear reason why it's difficult, I would prefer this in the form, `let (line1, line2) = match self.split_direction { ...`, and then just have each branch of the match statement resolve to a 2-tuple of lines.

Reply:
(to clarify: this isn't blocking, I'm just stating my preferences)

Reply:
I'm more than open to this kind of style nit - I've done a lot of study and only a little hobby work in rust. I'll see if I can work this the way you suggest.

Reply:
I made this change, here and in several other (child1_xxx, child2_xxx) locations. Definitely feels better.